src/figuresClasses.ts

```diff
@@ -1,19 +1,75 @@
+type Color = 'red' | 'blue' | 'green';
+type Shape = 'triangle' | 'cirlce' | 'rectangle';
+
 export interface Figure {
+  shape: Shape;
+  color: Color;
 
+  getArea(): number;
 }
 
-export class Triangle {
+export class Triangle implements Figure {
+  shape: Shape = 'triangle';
 
-}
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (this.a <= 0 && this.b <= 0 && this.c <= 0) {
+      throw new Error('Invalid element length');
+    }
 
-export class Circle {
+    if (a >= b + c || b >= a + c || c >= b + a) {
```

Comment:
I recommend you to use more array methods and variables for validation instead of multiple conditions inside `if` statement.